numpy/lib/_datasource.py

```diff
@@ -337,8 +337,8 @@ def _cache(self, path):
                 with closing(urlopen(path)) as openedurl:
                     with _open(upath, 'wb') as f:
                         shutil.copyfileobj(openedurl, f)
-            except URLError:
-                raise URLError("URL not found: %s" % path)
+            except URLError as e:
```

Comment:
Can you show what the resulting error message is with this change?

Reply:
- *Before invalid URL exeption ( I run this on my personal conda environment ) was*

```
Traceback (most recent call last):
  File "/home/sarthak/anaconda3/lib/python3.7/site-packages/numpy/lib/_datasource.py", line 418, in _cache
    with closing(urlopen(path)) as openedurl:
  File "/home/sarthak/anaconda3/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/home/sarthak/anaconda3/lib/python3.7/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/home/sarthak/anaconda3/lib/python3.7/urllib/request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "/home/sarthak/anaconda3/lib/python3.7/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/home/sarthak/anaconda3/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/home/sarthak/anaconda3/lib/python3.7/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sarthak/anaconda3/lib/python3.7/site-packages/numpy/lib/_datasource.py", line 422, in _cache
    raise URLError("URL not found: %s" % path)
urllib.error.URLError: <urlopen error URL not found: https://raw.githubusercontent.com/numpy/numpy/master/toxsadas.ini>
```
- *After the change*
```
Traceback (most recent call last):
  File "/home/sarthak/Desktop/projects/numpy/numpy/lib/_datasource.py", line 337, in _cache
    with closing(urlopen(path)) as openedurl:
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sarthak/Desktop/projects/numpy/numpy/lib/_datasource.py", line 341, in _cache
    raise URLError("URL not found: %s" % path) from e
urllib.error.URLError: <urlopen error URL not found: https://raw.githubusercontent.com/numpy/numpy/master/toxsadas.ini>
```